js_modules/dagster-ui/packages/ui-core/src/assets/LaunchAssetExecutionButton.tsx

```diff
@@ -332,14 +337,28 @@ export const useMaterializationAction = (preferredJobName?: string) => {
     }
     setState({type: 'loading'});
 
-    const data = await onLoad(assetKeysOrJob);
+    let data = await onLoad(assetKeysOrJob);
 
     if ('assetNodeDefinitionCollisions' in data && data.assetNodeDefinitionCollisions.length) {
       showCustomAlert(buildAssetCollisionsAlert(data));
       setState({type: 'none'});
       return;
     }
 
+    if ('assetNodeAdditionalRequiredKeys' in data && data.assetNodeAdditionalRequiredKeys.length) {
+      setState({type: 'none'});
+      try {
+        await confirm(buildAssetAdditionalRequiredKeysAlert(data));
+
+        if (assetKeysOrJob instanceof Array) {
+          data = await onLoad([...assetKeysOrJob, ...data.assetNodeAdditionalRequiredKeys]);
+        }
+        setState({type: 'loading'});
```

Comment:
If we're `awaiting` here do we actually end up seeing the `loading` state? Seems like the loading state happens after the data is already returned?

Reply:
Good catch this was wrong 🙏